src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -214,10 +214,12 @@ def _import_file(
             file_path, self._session._conn.stage_registry, import_path
         )
 
-        if sproc_name:
-            self._sproc_level_imports[sproc_name].add(absolute_module_path)
-        else:
-            self._session_level_imports.add(absolute_module_path)
+        if absolute_module_path not in sys.path:
```

Comment:
I'm a bit confused by how this change is related to the tests in `tests/mock_unit/test_udf.py` but I see the issue. However this should be resolved using the same approach as `frozen_sys_module_keys`, i.e. saving a copy of `sys.path` before processing imports and restore in the exit stack.

Reply:
https://github.com/snowflakedb/snowpark-python/blob/93b32a1ef04181c56b18e02441af6ffdb4b2dfce/src/snowflake/snowpark/mock/_stored_procedure.py#L127C9-L127C31

Reply:
updated the implementation, thanks for pointing